app/src/main/java/com/everest/userinfo/validations/ValidateFields.kt

```diff
@@ -0,0 +1,65 @@
+package com.everest.userinfo.validations
+
+
+import android.util.Patterns
+import android.widget.Toast
+import com.everest.userinfo.HomeActivity
+import com.everest.userinfo.databinding.ActivityHomeBinding
+
+class ValidateFields constructor(
+    private var binding: ActivityHomeBinding,
+    private var homeActivity: HomeActivity
```

Comment:
`ValidateFields` is tightly coupled to `HomeActivity`. We can make it loosly coupled.
You can take AppCompatActivity or Activity instead of HomeActivity